torch/onnx/symbolic_opset9.py

```diff
@@ -6629,3 +6629,24 @@ def onnx_placeholder(g: jit_utils.GraphContext, *inputs, **attrs):
     env = g.env
 
     return torch._C._jit_onnx_convert_pattern_from_subblock(block, node, env)
+
+
+@_onnx_symbolic("aten::_conj")
+@_onnx_symbolic("aten::resolve_conj")
+@_onnx_symbolic("aten::resolve_neg")
+@_beartype.beartype
+def unsupported_complex_operators(g: jit_utils.GraphContext, input: _C.Value):
+    # ONNX does not have operators to manipulate real/imaginary components directly
+    # However, some torch operations use complex operations even when input is real,
+    # which results in failures due to missing operators for complex numbers
+    # e.g. `print(torch.randn(10, 5))` calls aten::resolve_conj, aten::_conj and aten::resolve_neg
+    #   although input is float
+
+    # This symbolic is a no-op for real numbers and raise a an exception when input is complex
```

Comment:
Like https://github.com/pytorch/pytorch/pull/86180 pointed out, it is always a noop for onnx because this op is just manipulating internal storage. Also see:

```
In [1]: import torch

In [2]: x = torch.tensor([-1 + 1j, -2 + 2j, 3 - 3j])

In [3]: x
Out[3]: tensor([-1.+1.j, -2.+2.j,  3.-3.j])

In [4]: y = x.conj()

In [5]: y
Out[5]: tensor([-1.-1.j, -2.-2.j,  3.+3.j])

In [6]: y.is_conj()
Out[6]: True

In [7]: z = y.resolve_conj()

In [8]: z
Out[8]: tensor([-1.-1.j, -2.-2.j,  3.+3.j])

In [9]: z.is_conj()
Out[9]: False

In [10]: y == z
Out[10]: tensor([True, True, True])
```

Reply:
What is the recommendation for the comment above?

Complex numbers will never reach the symbolics for resolve_conj, _conj or _neg. The proposed no-ops symbolics allow the graph to be produced and optimize what away dead nodes such as theses

Reply:
The recommendation is to treat everything as noop and not raise errors. If complex numbers never reach these symbolics, could you say more on why we raise exception when input is complex? I may be missing something here

Reply:
from lines 6647-6648
```
        # _C._jit_pass_onnx_node_shape_type_inference raises when shape_inference is enabled,
        # but this call make sure an  exception still raises without shape inference
```

Reply:
aten::_conj does seem to need an error if onnx doesn't do conjugation right now; do we need errors for the other two ops?

Reply:
The three symbolics are needed to handle either `print(torch.randn(1,1))`, `print(torch.randn(1,1,dtype=torch.cfloat))`, `print(torch.randn(1,1).conj())`, `print(torch.randn(1,1).resolve_conj())`

It goes beoynd the repro provided in the issue, but cover all possibilities of having unexpected complex numbers reaching the exporter